tests/integration/__init__.py

```diff
@@ -0,0 +1,16 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
```

Comment:
I think we do not need `__init__.py` in tests folder.

Reply:
I would agree with you, but it raises an error:

```
_____________________________________________________________________________ ERROR collecting tests/integration/test_hive.py ______________________________________________________________________________
import file mismatch:
imported module 'test_hive' has this __file__ attribute:
  /Users/fokkodriesprong/Desktop/iceberg-python/tests/catalog/test_hive.py
which is not the same as the test file we want to collect:
  /Users/fokkodriesprong/Desktop/iceberg-python/tests/integration/test_hive.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
_____________________________________________________________________________ ERROR collecting tests/integration/test_rest.py ______________________________________________________________________________
import file mismatch:
imported module 'test_rest' has this __file__ attribute:
  /Users/fokkodriesprong/Desktop/iceberg-python/tests/catalog/test_rest.py
which is not the same as the test file we want to collect:
  /Users/fokkodriesprong/Desktop/iceberg-python/tests/integration/test_rest.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
```

We have the `__init__.py` files in the other folders as well 🤔

Reply:
I raised this because we removed all `__init__.py` in tests a year ago: https://github.com/apache/iceberg/pull/5919
I think we can rename these two files to avoid name collisions with those in `catalogs/`. This would enable us to safely delete the `__init__.py`

The other `__init__.py` in `tests/` was added in https://github.com/apache/iceberg-python/pull/45/. I removed this one and all tests still work.

It would be ideal to maintain consistency by not having `__init__.py` files in test folders. However, I believe this is a minor issue and should not block this PR.